src/main/scala/com/amazon/deequ/analyzers/QuantileNonSample.scala

```diff
@@ -0,0 +1,307 @@
+    val rankMap = getRankMap()
+    val tmp = rankMap.keySet.toArray
+    val totalWeight = rankMap.last._2
+    var ret = ArrayBuffer[(T,Double)]()
+    tmp.foreach{ item =>
+      ret = ret :+ (item, rankMap(item).toDouble / totalWeight.toDouble)
+    }
+    ret.toArray
+  }
+
+  /**
+   * Get the rank of query item (inclusive rank) without RankMap.
+   * @param item item to query
+   * @return the estimated rank of the query item in sketch
+   */
+  def getRank(item:T): Long = {
+    var r = 0L
+    output.foreach(tuple => {
+      if (!compareGreater(tuple._1,item)) {
+        r = r + tuple._2
+      }
+    })
+    r
+  }
+
+  /**
+   * Get the rank of query item (exclusive rank) without RankMap.
+   * @param item item to query
+   * @return the estimated rank of the query item in sketch
+   */
+  def getRankExclusive(item:T): Long = {
+    var r = 0L
+    output.foreach(tuple => {
+      if (compareSmaller(tuple._1,item)) {
+        r = r + tuple._2
+      }
+    })
+    r
+  }
+
+  /**
+   * Get the rank of query item with RankMap.
+   * @param item item to query
+   * @param rankMap the estimated rank of the query item in sketch
+   * @return
+   */
+  def getRank(item:T, rankMap:ListMap[T,Long]): Long = {
+    var curRank = 0L
+    breakable {
+      for (target <- rankMap) {
+        if (compareGreater(target._1, item))
+          break
+        curRank = target._2
+      }
+    }
+    curRank
+  }
+
+
+
+  /**
+   * merge two sketches into a single one
+   *
+   * @param that another sketch
+   * @return the merged sketch
+   */
+  def merge(that: QuantileNonSample[T]) : QuantileNonSample[T] = {
+    while (this.curNumOfCompactors < that.curNumOfCompactors) {
+      this.expand()
+    }
+
+    for (i <- 0 until that.curNumOfCompactors) {
+      this.compactors(i).buffer = this.compactors(i).buffer ++ that.compactors(i).buffer
+    }
+
+    compactorActualSize = getCompactorItemsCount
+
+    while (compactorActualSize >= compactorTotalSize) {
+      this.condense()
+    }
+    this
+  }
+
+  private def output: Array[(T,Long)] = {
+    compactors.toArray.slice(0,curNumOfCompactors).zipWithIndex.flatMap{case (compactor,i) =>
+      compactor.buffer.toArray.map((_,1L << i))
+    }
+  }
+
+  /**
+   * the quantile values of the sketch
+   *
+   * @param q number of quantiles required
+   * @return quantiles 1/q through (q-1)/q
+   */
+  def quantiles(q: Int) : Array[T] = {
+    val sortedItems = output.sortBy(_._1)
+    val size = sortedItems.map(_._2).sum
+    var nextThresh = size/q
+    var curq = 1
+    var i=0
+    var sumSoFar:Long=0
+    val quantiles = Array.fill[T](q-1)(sortedItems(0)._1)
+
+
+    while (i<sortedItems.length && curq<q) {
+      while (sumSoFar < nextThresh) {
+        sumSoFar += sortedItems(i)._2
+        i+=1
+      }
+      quantiles(curq -1) = sortedItems(math.min(i,sortedItems.length-1))._1
+      curq +=1
+      nextThresh = curq * size / q
+    }
+    quantiles
+  }
+
+  /**
```

Comment:
Please format the rest of this file according to our [coding conventions](https://github.com/databricks/scala-style-guide)

Reply:
Added comments for public functions.